desc/io/input_reader.py

```diff
@@ -686,18 +686,26 @@ def write_desc_input(filename, inputs, header=""):
         f.write("\n# continuation parameters\n")
         for key in ["bdry_ratio", "pres_ratio", "curr_ratio", "pert_order"]:
             f.write(
-                key + " = {} \n".format(", ".join([str(inp[key]) for inp in inputs]))
+                key
+                + " = {} \n".format(", ".join([str(float(inp[key])) for inp in inputs]))
             )
 
         f.write("\n# solver tolerances\n")
         for key in ["ftol", "xtol", "gtol", "nfev"]:
+            inputs_not_None = []
+            for inp in inputs:
+                if inp[key] is not None:
+                    inputs_not_None.append(inp)
+            if not inputs_not_None:  # an  empty list evals to False
+                break  # don't write line if all input tolerance are None
```

Comment:
I think we want `continue` (skip to next iteration) here instead of `break` (exit the loop entirely)

Reply:
you're right, I added a test for this now (gtol will not be written but rest will be)